cmake/modules/BuildRocksDB.cmake

```diff
@@ -1,4 +1,5 @@
 function(build_rocksdb)
+  find_program (MAKE_EXECUTABLE NAMES make gmake)
```

Comment:
I would expect this to find `make` before `gmake` 
Which will result in using the wrong make on FreeBSD.
Reversing the order would avoid this

Reply:
@ideepika could you please fix BuildSPDK.cmake as suggested by Willem in another commit in this PR?

Reply:
@wjwithagen @tchaikov made the necessary changes.